components/scoreHandler.ts

```diff
@@ -882,11 +883,20 @@ export async function getMissionEndData(
 
     const newLocationXp = completionData.XP
     let newLocationLevel = levelForXp(newLocationXp, masteryData?.XpPerLevel)
+    const userProgressionLocations = userData.Extensions.progression.Locations
 
     if (!query.masteryUnlockableId) {
-        userData.Extensions.progression.Locations[
-            locationParentId
-        ].PreviouslySeenXp = newLocationXp
+        if (userProgressionLocations[locationParentId]) {
+            userProgressionLocations[locationParentId].PreviouslySeenXp = newLocationXp
+        } else {
+            log(LogLevel.WARN, `Location progression missing for ${locationParentId}, adding default progression.`)
+            const defaultProgression: ProgressionData = {
+                Xp: 0,
+                Level: 1,
+                PreviouslySeenXp: newLocationXp
```

Comment:
```suggestion
        userProgressionLocations[locationParentId] ??= {
            Xp: 0,
            Level: 1,
            PreviouslySeenXp: newLocationXp
        }
        userProgressionLocations[locationParentId].PreviouslySeenXp = newLocationXp
```

per grape's suggestion